### DISCUSSION TITLE
`$TURBO_ROOT$` special string

### DISCUSSION THREAD

**Discussion by abernier** [vercel/turborepo#8877] [Ideas]
### Goals

Since `turbo.json.tasks.inputs`' paths are relative to the package, will avoid `../../..` to locate a file outside 

### Background

https://discord.com/channels/818588653005176832/1267594181015244821

### Proposal

before:

```json
{
  "tasks": {
    "test": {
      "inputs": [
        "$TURBO_DEFAULT$",
        "dist/**",
        "../../packages/e2e/bin/test.mjs",
        "../../packages/e2e/{playwright.config.ts,snapshot.test.js}"
      ],
      "cache": true
    },
    ...
  }
}
```

after:

```json
{
  "tasks": {
    "test": {
      "inputs": [
        "$TURBO_DEFAULT$",
        "dist/**",
        "$TURBO_ROOT$/packages/e2e/bin/test.mjs",
        "$TURBO_ROOT$/packages/e2e/{playwright.config.ts,snapshot.test.js}"
      ],
      "cache": true
    },
    ...
  }
}
```

**Comment by anthonyshew** [comment]
Can you add to the original post the problem that this solves? Adding specialized microsyntaxes comes with the cost of users having specialized knowledge of Turborepo. While I'm sure this potentially does solve a problem, `../../` doesn't immediately strike me as a pain point by itself.

What problem are we solving as we trade the simplicity of universal path patterns (`../../`) for a Turborepo-specific micro-syntax?

**Comment by abernier** [comment]
maybe it's not an issue then, was just surprised to go relative here to point out files from another package, but why not

(or maybe using the name of another package is already possible? `@my/otherpkg/path/to/file1`)

**Reply by anthonyshew** [reply]
I wasn't necessarily saying it's not a worthwhile idea. I want to understand more where this idea came from and what we want to solve with it. 😄 

P.S. No, you're not able to use a package name for these globs at this time.

**Comment by wagenet** [comment]
Relative paths are fragile and could fail if the nesting varies between workspace packages.

**Reply by anthonyshew** [reply]
This was what I was thinking of and was hoping to see added to the original post. I didn't want to suggest a use case in case that wasn't the one OP was looking for. 😁

**Comment by jzxhuang** [comment]
would love to see something like this! Gave a concrete use case here: https://github.com/vercel/turborepo/discussions/9203
